zulipterminal/ui_tools/boxes.py

```diff
@@ -153,6 +155,13 @@ def autocomplete_emojis(self, text: str) -> Tuple[List[str], List[str]]:
         return emojis, emoji_typeahead
 
     def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
+        if self.typeahead_mode:
```

Comment:
Without this commit, what is the expected behavior? The commit describes the new behavior, but not the previous - is this an enhancement on the footer code, or a requirement for it to work 'properly'?

Reply:
I've explained this in a comment in the discussion above, but I think I'll add it to the commit text too.
This commit fixes the problem of the footer continuing to display the suggestions once user is done with auto-complete. Once the footer is set, there's no way of knowing 'when' to reset the footer. Ideally this should be squashed with the previous commit while merging, but I've kept them seperate so there is a little distinction while reviewing.

Reply:
This does make the commit rather big; I'm not sure at this point how to split it, as it seems non-ideal right now to have it stay in the autocomplete state between commits.

Reply:
I'm mostly happy with this separation at this point, though leaving the UI as it is in the previous commit just seems a bit messy.

Reply:
I could squash before merging?

Reply:
It's fine to leave for now.